src/NServiceBus.Core.Tests/Unicast/Messages/DefaultMessageRegistryTests.cs

```diff
@@ -47,9 +47,20 @@ public void Should_return_the_correct_parent_hierarchy()
                 Assert.AreEqual(typeof(ConcreteParent1), messageMetadata.MessageHierarchy.ToList()[2]);
                 Assert.AreEqual(typeof(InterfaceParent1Base), messageMetadata.MessageHierarchy.ToList()[3]);
                 Assert.AreEqual(typeof(ConcreteParentBase), messageMetadata.MessageHierarchy.ToList()[4]);
-
             }
 
+            [TestCase("NServiceBus.Unicast.Tests.DefaultMessageRegistryTests+When_getting_message_definition+MyEvent, NonExistingAssembly, Version=1.0.0.0, Culture=neutral, PublicKeyToken=b50674d1e0c6ce54")]
+            [TestCase("NServiceBus.Unicast.Tests.DefaultMessageRegistryTests+When_getting_message_definition+MyEvent, NonExistingAssembly, Version=1.0.0.0, Culture=neutral")]
+            [TestCase("NServiceBus.Unicast.Tests.DefaultMessageRegistryTests+When_getting_message_definition+MyEvent, NonExistingAssembly")]
+            public void Should_match_types_from_a_different_assembly(string typeName)
```

Comment:
Shouldn't there be one more test case here? The one without the assembly name?

Reply:
done